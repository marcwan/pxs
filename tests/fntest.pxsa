MODULE_BODY:
        DECLAREFN       factorial, ::__FACTORIAL_123__::
        DECLARE         res
;        PUSHARG         10
        CALLFUNC        factorial
        SET             res, ::__RETURN_VALUE__::
        DEBUGPRINT      res
        EXIT            0
END_MODULE_BODY

FUNCTION ::__FACTORIAL_123__::
        DECLARE         idx
        SET             idx, 4
;        POPARG          idx
        DECLARE         result
        SET             result, 1
        DECLARE         i
        SET             i, 0
__FOR_LOOP_12345_TEST:
        DECLARE         ::__TMP1_::
        SUB             idx, 1, ::__TMP1_::
        COMPARE         i, ::__TMP1_::
        REMOVEVAR       ::__TMP1_::
        JUMPGTE         __FOR_LOOP_12345_END
        DECLARE         ::__TMP2__::
        SUB             idx, i, ::__TMP2__::
        MUL             result, ::__TMP2__::, result
__FOR_LOOP_12345_INCR:
        ADD             i, 1, i
        JUMP            __FOR_LOOP_12345_TEST
__FOR_LOOP_12345_END:
        DEBUGPRINT      result
        EXITFUNC        result
END_FUNCTION ::__FACTORIAL_123__::
