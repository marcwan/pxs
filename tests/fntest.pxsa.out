DEBUGPRINT: result = 24.000000
DEBUGPRINT: res = 24.000000
